### ISSUE TITLE
Vue: Use `parameters.component` if available

### ISSUE THREAD

**Issue by mattgreenfield** [storybookjs/storybook#8109]
**Is your feature request related to a problem? Please describe.**
When writing stories in Vue, each story must declare the component that it displays. This leads to a lot of repetition of the component name. 

```js
import SButton from './SButton.vue';

export default { title: 'Button' };

export const loading = () => ({
  components: { SButton }, // This is the same on every story
  template: '<s-button loading>Manage</s-button>',
});
export const small = () => ({
  components: { SButton }, // This is the same on every story
  template: '<s-button size="small">Manage</s-button>',
});
```

This is true of CSF and StoriesOf stories but I noticed that the default export of a CSF story has a `component` option. I had assumed that this would register the component globally (or scoped to the file?) and so I wouldn't need to register it for each story. Like this;
`export default { title: 'Button', component: SButton };`
But unfortunately this didn't work as expected.

**Describe the solution you'd like**
Declaring a component on the default export means you don't have to add it to every story.
```js
import SButton from './SButton.vue';

export default { title: 'Button', component: SButton };

export const loading = () => ({
  template: '<s-button loading>Manage</s-button>',
});
export const small = () => ({
  template: '<s-button size="small">Manage</s-button>',
});
```

**Describe alternatives you've considered**
I considered making a small wrapper function at the top of my story file to do this for me. Something along the lines of;
```js
const storyMaker = (template) => {
  return {
    components: { SButton },
    template: template,
  }
};

export const loading = () => storyMaker('<s-button loading>Manage</s-button>');
```

**Are you able to assist bring the feature to reality?**
no

**Additional context**
Originally raised on Medium - https://medium.com/@matt.greenfield/this-looks-great-and-much-easier-to-use-thank-you-c560cf9865ba
Issue created as per response to that post.

**Comment by pksunkara** [comment]
@Aaron-Pool I like this, What do you think? I saw that you used something like this in https://github.com/Aaron-Pool/Vue-Cli-Storybook-Docs/blob/master/src/stories/index.stories.js#L9

**Comment by ux-engineer** [comment]
@shilman this is listed in **6.0 priority features** in Hotlist, but is included in **7.0 milestone**?

**Comment by shilman** [comment]
We’re cleaning house! Storybook has changed a lot since this issue was created and we don’t know if it’s still valid. Please open a new issue referencing this one if this is still relevant in SB 7.x.
